game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleActions.java

```diff
@@ -23,6 +23,15 @@ void queryRetreat(
 
   void fireNavalBombardment(IDelegateBridge bridge);
 
+  void landParatroopers(
+      IDelegateBridge bridge, Collection<Unit> airTransports, Collection<Unit> dependents);
+
+  void removeNonCombatants(IDelegateBridge bridge);
+
+  void markNoMovementLeft(IDelegateBridge bridge);
```

Comment:
At this point, I'm not trying to understand what each of these methods do.  My goal is that once all of the steps have been converted, I'd like to go through these methods and try to "shrink" their focus by splitting them, breaking them up, etc and adding tests to them.  Some of them do a lot of things (queryRetreat has >120 lines).

At that point, I think I'll understand the methods well enough to be able to write javadocs for each of them.

Reply:
Another part of my goal is to have all of the code that interacts with the battle state to be limited to the methods in `BattleState`.  Some of the methods in `BattleActions` interact with the battle state and so I'll need to separate them.  Once that is done, it is possible that `BattleActions` won't even be needed anymore since the `IDelegateBridge` can be interacted with in the steps just as well as it can be interacted in `BattleActions`.  But at this moment, I'm just trying to move things around to make it easier to see how the logic should be split up.